server/src/test/scala/org/http4s/server/middleware/authentication/AuthMiddlewareSpec.scala

```diff
@@ -76,6 +77,90 @@ class AuthMiddlewareSpec extends Http4sSpec {
       service.orNotFound(Request[IO](method = Method.POST)) must returnStatus(Ok)
       service.orNotFound(Request[IO](method = Method.GET)) must returnStatus(NotFound)
     }
+
+    "return 404 for an unmatched but authenticated route" in {
+      val userId: User = 42
+
+      val authUser: Kleisli[OptionT[IO, ?], Request[IO], User] =
+        Kleisli.pure(userId)
+
+      val authedService: AuthedService[User, IO] =
+        AuthedService {
+          case POST -> Root as _ => Ok()
+        }
+
+      val middleware = AuthMiddleware(authUser)
+
+      val service = middleware(authedService)
+
+      service.orNotFound(Request[IO](method = Method.POST)) must returnStatus(Ok)
```

Comment:
Is this to test your test?

Reply:
None of the prior tests actually test the default apply in `authmiddleware`. They test the second apply with explicit error.

I figured it doesn't hurt to test whether it passes through anyway.

Reply:
I moved this to a separate test, actually.